openfl-tutorials/interactive_api/PyTorch_BrainMRI_UNet/prepare_dataset.py

```diff
@@ -0,0 +1,128 @@
+    inp = input()
+    if inp=='y' or inp=='Y':
+        pass
+    else:
+        sys.exit("Exiting program")
+else:
+    os.mkdir(img_path_new)
+
+
+msk_path_new = save_folder + 'msk/'
+if os.path.exists(msk_path_new):
+    print("processed_data/msk/ already exists, do you want to proceed? press y/Y to proceed and n/N to cancel")
+    inp = input()
+    if inp=='y' or inp=='Y':
+        pass
+    else:
+        sys.exit("Exiting program")
+else:
+    os.mkdir(msk_path_new)
+
+
+# Two CSV files 'train_data.csv' and 'valid_data.csv' are created which will have the maps
+train_data_csv = save_folder + 'train_data.csv'
+valid_data_csv = save_folder + 'valid_data.csv'
+with open(train_data_csv,'w') as f:
+    f.write('name,img_path,msk_path\n')
+with open(valid_data_csv,'w') as f:
+    f.write('name,img_path,msk_path\n')
+
+# process the train images
+for t in tqdm(train_image_list):
+    img_name = t['image'][t['image'].rfind('/')+1:t['image'].rfind('.nii')]
+    img_path = os.path.join(data_folder, t['image'])
+    msk_path = os.path.join(data_folder, t['label'])
+    img = np.array(nib.load(img_path).dataobj)[:,:,:,0]
+    msk = np.array(nib.load(msk_path).dataobj)
+    num_slices = msk.shape[2]
+    for slice in range(num_slices):
+        img_slice = img[:,:,slice]
+        msk_slice = msk[:,:,slice]
+        img_slice_name = os.path.join(img_path_new,f'{img_name}_{slice}.npy')
+        msk_slice_name = os.path.join(msk_path_new,f'{img_name}_{slice}.npy')
+        np.save(img_slice_name,img_slice)
+        np.save(msk_slice_name,msk_slice)
+        with open(train_data_csv,'a') as f:
+            f.write(f'{img_name},{img_slice_name},{msk_slice_name}\n')
+
+# process the validation images
+for v in tqdm(valid_image_list):
+    img_name = v['image'][v['image'].rfind('/')+1:v['image'].rfind('.nii')]
+    img_path = os.path.join(data_folder, v['image'])
+    msk_path = os.path.join(data_folder, v['label'])
+    img = np.array(nib.load(img_path).dataobj)[:,:,:,0]
+    msk = np.array(nib.load(msk_path).dataobj)
+    num_slices = msk.shape[2]
+    for slice in range(num_slices):
+        img_slice = img[:,:,slice]
+        msk_slice = msk[:,:,slice]
+        img_slice_name = os.path.join(img_path_new,f'{img_name}_{slice}.npy')
+        msk_slice_name = os.path.join(msk_path_new,f'{img_name}_{slice}.npy')
+        np.save(img_slice_name,img_slice)
+        np.save(msk_slice_name,msk_slice)
+        with open(valid_data_csv,'a') as f:
+            f.write(f'{img_name},{img_slice_name},{msk_slice_name}\n')
+
+# Copy the dataset to the respective folders
+os.system('cp -r processed_data envoy_1/')
+os.system('cp -r processed_data envoy_2/')
+os.system('cp processed_data/train_data.csv envoy_1/')
+os.system('cp processed_data/train_data.csv envoy_2/')
+os.system('cp processed_data/valid_data.csv envoy_1/')
+os.system('cp processed_data/valid_data.csv envoy_2/')
```

Comment:
@saransh09 , can we refactor code to add the following at the end of the file:
```python3
if __name__ == '__main__':
    main()
```

Reply:
> @saransh09 , can we refactor code to add the following at the end of the file?
> 
> ```python
> if __name__ == '__main__':
>     main()
> ```

Yes sure, I will do this, on it.